### ISSUE TITLE
frida-gumjs devkit crash

### ISSUE THREAD

**Issue by Cookgo** [frida/frida-gum#1030]
below i just have copy and so simply modify the frida-gumjs-example.c,performing  test_fork twice.The program be crashed, and i have test the stalker and pure c api hook(attach),works too.Only gumjs crash, is v8's problem? If there are pure c api which can hook art , something would be easier,but so a big pity,the Java Hooking highly bingding with js engine 
 
static void on_message (const gchar * message, GBytes * data, gpointer user_data);

void test_gumjs()
{
    GumScriptBackend * backend;
    GCancellable * cancellable = NULL;
    GError * error = NULL;
    GumScript * script;
    GMainContext * context;

    gum_init_embedded ();

    backend = gum_script_backend_obtain_qjs();

    script = gum_script_backend_create_sync (backend, "example",
                                             "Interceptor.attach(Module.getExportByName(null, 'open'), {\n"
                                             "  onEnter(args) {\n"
                                             "    console.log(`[*] open(\"${args[0].readUtf8String()}\")`);\n"
                                             "  }\n"
                                             "});\n"
                                             "Interceptor.attach(Module.getExportByName(null, 'close'), {\n"
                                             "  onEnter(args) {\n"
                                             "    console.log(`[*] close(${args[0].toInt32()})`);\n"
                                             "  }\n"
                                             "});",
                                             NULL, cancellable, &error);
    g_assert (error == NULL);

    gum_script_set_message_handler (script, on_message, NULL, NULL);

    gum_script_load_sync (script, cancellable);

    close (open ("/etc/hosts", O_RDONLY));
    close (open ("/etc/fstab", O_RDONLY));

    context = g_main_context_default ();
    while (g_main_context_pending (context))
        g_main_context_iteration (context, FALSE);

    gum_script_unload_sync (script, cancellable);

    g_object_unref (script);

    gum_deinit_embedded ();

}

static void
on_message (const gchar * message,
            GBytes * data,
            gpointer user_data)
{
    JsonParser * parser;
    JsonObject * root;
    const gchar * type;

    parser = json_parser_new ();
    json_parser_load_from_data (parser, message, -1, NULL);
    root = json_node_get_object (json_parser_get_root (parser));
    type = json_object_get_string_member (root, "type");
    if (strcmp (type, "log") == 0)
    {
        const gchar * log_message;

        log_message = json_object_get_string_member (root, "payload");
        LOGD ("%s", log_message);
    }
    else
    {
        LOGD ("on_message: %s\n", message);
    }

    g_object_unref (parser);
}

void test_fork()
{
   test_gumjs();
   sleep(2);
    test_gumjs();

//    int pid=fork();
//    if(pid==0){
//        // 子进程
//        LOGD("child pid:%d",getpid());
//        test_gumjs();
//
//    }
//    else{
//        // 父进程
//        LOGD("parent pid:%d",getpid());
//
//    }

}

__attribute__((constructor))
static void main_entry ()
{

 
    std::thread t(test_fork);
    t.detach();
}

--------- beginning of system
--------- beginning of main
2025-05-08 04:20:47.093 27123-27123 giao                    com.agent                            D  onClick: 
2025-05-08 04:20:48.641 27123-27123 nativeloader            com.agent                            D  Load /data/app/~~wpHHeujnWR0oi_QVks1Nuw==/com.agent-84Zl6aXzWNjO6W9HD1Df-w==/base.apk!/lib/arm64-v8a/libdlib.so using ns clns-7 from class loader (caller=/data/app/~~wpHHeujnWR0oi_QVks1Nuw==/com.agent-84Zl6aXzWNjO6W9HD1Df-w==/base.apk!classes4.dex): ok
2025-05-08 04:20:48.646 27123-27123 Choreographer           com.agent                            I  Skipped 93 frames!  The application may be doing too much work on its main thread.
2025-05-08 04:20:48.658 27123-27515 giao                    com.agent                            D  on_message: [*] close(129)
2025-05-08 04:20:48.658 27123-27515 giao                    com.agent                            D  on_message: [*] close(-1)
2025-05-08 04:20:48.664 27123-27154 HWUI                    com.agent                            I  Davey! duration=1565ms; Flags=0, FrameTimelineVsyncId=3245157, IntendedVsync=15889340301256, Vsync=15890892133759, InputEventId=-134860247, HandleInputStart=15890896981513, AnimationStart=15890896983710, PerformTraversalsStart=15890902821031, DrawStart=15890902886460, FrameDeadline=15889373587627, FrameInterval=15890896662543, FrameStartTime=16672737, SyncQueued=15890904101955, SyncStart=15890904168443, IssueDrawCommandsStart=15890904222561, SwapBuffers=15890904728379, FrameCompleted=15890905370632, DequeueBufferDuration=17985, QueueBufferDuration=207967, GpuCompleted=15890905370632, SwapBuffersCompleted=15890905021592, DisplayPresentTime=0, CommandSubmissionCompleted=15890904728379, 
--------- beginning of crash
2025-05-08 04:20:50.666 27123-27515 libc                    com.agent                            A  FORTIFY: pthread_mutex_lock called on a destroyed mutex (0x72c41b32b0)
2025-05-08 04:20:50.667 27123-27515 libc                    com.agent                            A  Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 27515 (com.agent), pid 27123 (com.agent)
2025-05-08 04:20:50.974 27519-27519 DEBUG                   pid-27519                            A  Cmdline: com.agent
2025-05-08 04:20:50.974 27519-27519 DEBUG                   pid-27519                            A  pid: 27123, tid: 27515, name: com.agent  >>> com.agent <<<
2025-05-08 04:20:50.974 27519-27519 DEBUG                   pid-27519                            A        #04 pc 0000000002326ac0  /data/app/~~wpHHeujnWR0oi_QVks1Nuw==/com.agent-84Zl6aXzWNjO6W9HD1Df-w==/base.apk!libdlib.so (offset 0x5954000) (BuildId: 07dd3fb7dbb3b583b95cb6fd54068d638c2ba9ff)
2025-05-08 04:20:50.974 27519-27519 DEBUG                   pid-27519                            A        #05 pc 0000000002326e24  /data/app/~~wpHHeujnWR0oi_QVks1Nuw==/com.agent-84Zl6aXzWNjO6W9HD1Df-w==/base.apk!libdlib.so (offset 0x5954000) (BuildId: 07dd3fb7dbb3b583b95cb6fd54068d638c2ba9ff)
2025-05-08 04:20:50.974 27519-27519 DEBUG                   pid-27519                            A        #06 pc 0000000002326db4  /data/app/~~wpHHeujnWR0oi_QVks1Nuw==/com.agent-84Zl6aXzWNjO6W9HD1Df-w==/base.apk!libdlib.so (offset 0x5954000) (_frida_g_private_get+8) (BuildId: 07dd3fb7dbb3b583b95cb6fd54068d638c2ba9ff)
2025-05-08 04:20:50.974 27519-27519 DEBUG                   pid-27519                            A        #07 pc 00000000022f17b8  /data/app/~~wpHHeujnWR0oi_QVks1Nuw==/com.agent-84Zl6aXzWNjO6W9HD1Df-w==/base.apk!libdlib.so (offset 0x5954000) (_frida_g_logv+368) (BuildId: 07dd3fb7dbb3b583b95cb6fd54068d638c2ba9ff)
2025-05-08 04:20:50.974 27519-27519 DEBUG                   pid-27519                            A        #08 pc 00000000022f162c  /data/app/~~wpHHeujnWR0oi_QVks1Nuw==/com.agent-84Zl6aXzWNjO6W9HD1Df-w==/base.apk!libdlib.so (offset 0x5954000) (_frida_g_log+88) (BuildId: 07dd3fb7dbb3b583b95cb6fd54068d638c2ba9ff)
2025-05-08 04:20:50.974 27519-27519 DEBUG                   pid-27519                            A        #09 pc 0000000001870214  /data/app/~~wpHHeujnWR0oi_QVks1Nuw==/com.agent-84Zl6aXzWNjO6W9HD1Df-w==/base.apk!libdlib.so (offset 0x5954000) (gum_init_embedded+140) (BuildId: 07dd3fb7dbb3b583b95cb6fd54068d638c2ba9ff)
2025-05-08 04:20:50.974 27519-27519 DEBUG                   pid-27519                            A        #10 pc 00000000017f612c  /data/app/~~wpHHeujnWR0oi_QVks1Nuw==/com.agent-84Zl6aXzWNjO6W9HD1Df-w==/base.apk!libdlib.so (offset 0x5954000) (test_gumjs()+40) (BuildId: 07dd3fb7dbb3b583b95cb6fd54068d638c2ba9ff)
2025-05-08 04:20:50.974 27519-27519 DEBUG                   pid-27519                            A        #11 pc 00000000017f63a0  /data/app/~~wpHHeujnWR0oi_QVks1Nuw==/com.agent-84Zl6aXzWNjO6W9HD1Df-w==/base.apk!libdlib.so (offset 0x5954000) (test_fork()+20) (BuildId: 07dd3fb7dbb3b583b95cb6fd54068d638c2ba9ff)
2025-05-08 04:20:50.974 27519-27519 DEBUG                   pid-27519                            A        #12 pc 00000000017f6d40  /data/app/~~wpHHeujnWR0oi_QVks1Nuw==/com.agent-84Zl6aXzWNjO6W9HD1Df-w==/base.apk!libdlib.so (offset 0x5954000) (BuildId: 07dd3fb7dbb3b583b95cb6fd54068d638c2ba9ff)
2025-05-08 04:20:50.974 27519-27519 DEBUG                   pid-27519                            A        #13 pc 00000000017f6cd8  /data/app/~~wpHHeujnWR0oi_QVks1Nuw==/com.agent-84Zl6aXzWNjO6W9HD1Df-w==/base.apk!libdlib.so (offset 0x5954000) (BuildId: 07dd3fb7dbb3b583b95cb6fd54068d638c2ba9ff)
2025-05-08 04:20:50.974 27519-27519 DEBUG                   pid-27519                            A        #14 pc 00000000017f67c4  /data/app/~~wpHHeujnWR0oi_QVks1Nuw==/com.agent-84Zl6aXzWNjO6W9HD1Df-w==/base.apk!libdlib.so (offset 0x5954000) (BuildId: 07dd3fb7dbb3b583b95cb6fd54068d638c2ba9ff)
2025-05-08 04:20:51.038  1683-1971  InputDispatcher         system_server                        E  channel 'a7bb46b com.agent/com.agent.MainActivity' ~ Channel is unrecoverably broken and will be disposed!
